drivers/flash/nrf_qspi_nor.c

```diff
@@ -699,7 +707,7 @@ static int qspi_nor_write(const struct device *dev, off_t addr,
 
 	if (size < 4U) {
 		res = write_sub_word(dev, addr, src, size);
-	} else if (((uintptr_t)src < CONFIG_SRAM_BASE_ADDRESS)) {
+	} else if ((uintptr_t)src < NRF_SOC_SRAM_BEGIN) {
```

Comment:
Why not just  use `if (!nrfx_is_in_ram(src))`?

Reply:
:( didn't know that it exist. @ioannisg Would it be OK what is suggested here ^^?

Reply:
Of course, that's what I suggested in the first place:

>This is an nRF driver, for a Cortex-M platform.
SRAM always starts after 0x20000000 - would this be enough to distinguish between core and system ram? Do we need to use DTS?

Reply:
However, this does not mean that adding the sram0, sram1 in DTS is wrong. Actually, it might be useful in other occasions. We might have to come back to this - you can leave it out for now.

Reply:
> We might have to come back to this - you can leave it out for now.

I'm assuming this means your ping from earlier is no longer relevant, thanks

Reply:
Actually, it is relevant since @nvlsianpu has kept the dts node definition in the PR

Reply:
> Actually, it is relevant since @nvlsianpu has kept the dts node definition in the PR

OK, since you asked, creating nodes in the app core DTSI that are not accessible by the core feels wrong to me.

DTSpec says:

>The reg property describes the address of the device’s resources within the address space defined by its parent bus.

The app core address space doesn't contain anything there, right? So there shouldn't be a DT node there. Just my opinion.

Reply:
Yeah, that's why i put it as a "recommendation"... I don't mind much (and I realized I created some noise to @nvlsianpu already, about it)

Reply:
> Yeah, that's why i put it as a "recommendation"... I don't mind much (and I realized I created some noise to @nvlsianpu already, about it)

I think it should be taken out unless there is a use case for it. But I won't block the PR.